src/modules/dashboard/dashboard.service.ts

```diff
@@ -0,0 +1,69 @@
+import * as qs from 'qs';
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { ShelterSearchPropsSchema } from 'src/shelter/types/search.types';
+import { SearchSchema } from 'src/types';
+import { ShelterSearch } from 'src/shelter/ShelterSearch';
+import { Cron, CronExpression } from '@nestjs/schedule';
+
+@Injectable()
+export class DashboardService {
+  constructor(private readonly prismaService: PrismaService) {}
+
+  @Cron(CronExpression.EVERY_6_HOURS)
```

Comment:
Estou um pouco confuso... faz sentido ter um @Cron aqui? Olhando por cima do código é só uma query e não tem nenhuma mutabilidade de dados.

Reply:
então nesse caso, a utilizaçao desse cron se foi devido a uma conversa de ideias sobre essa tela de dashboard onde vai estar dados importantes e que precisariam ser atualizados num periodo de tempo.

Reply:
+1 aqui.

Se a ideia for ter essas infos em um cache pra evitar sobrecarga no banco, faz um pouco mais de sentido rodar o cronjob (apesar de pessoalmente preferir a abordagem de atualização de cache apenas quando solicitado).

De qualquer forma o resultado do cron recorrente não parece estar indo pra nenhum lugar

Reply:
> +1 aqui.
> 
> Se a ideia for ter essas infos em um cache pra evitar sobrecarga no banco, faz um pouco mais de sentido rodar o cronjob (apesar de pessoalmente preferir a abordagem de atualização de cache apenas quando solicitado).
> 
> De qualquer forma o resultado do cron recorrente não parece estar indo pra nenhum lugar

mas ai no caso o ideal seria uma camada de cache na controller, que ao invalidar o cache vai retornar o dado atualizado não? Cron geralmente é mais usado quando tu envolve alguma mutabilidade de dado pois envolve um processamento de dados.

Reply:
opa boa noite, o cron foi retirado @rodrigooler por favor revisar o PR